ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -177,15 +188,29 @@ module API {
      * - A submodule of a module
      * - An attribute of an object
      */
-    bindingset[m]
-    bindingset[result]
-    Node getMember(string m) { result = this.getASuccessor(Label::member(m)) }
+    pragma[inline]
+    Node getMember(string m) { result = pragma[only_bind_out](this).getMemberInternal(m) }
+
+    /**
+     * INTERNAL USE ONLY.
+     *
+     * Same as `getMember` but without join-order hints.
+     */
+    cached
+    Node getMemberInternal(string m) {
```

Comment:
I would prefer to have such internal predicates be either top-level predicates or on a an internal `Impl` class.

Reply:
Moved to an internal class at `API::Node::Internal`.